### PR TITLE
Add interface for scoped keeper for 04-channel

### PR DESCRIPTION
**Pull request by Anmol1696**
## Description

Add scoped keeper interface for 04-channel module.

Part of: #2020 

---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [ ] Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#pr-targeting))
- [ ] Linked to Github issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Code follows the [module structure standards](https://github.com/cosmos/cosmos-sdk/blob/master/docs/building-modules/structure.md).
- [ ] Wrote unit and integration [tests](https://github.com/cosmos/ibc-go/blob/master/CONTRIBUTING.md#testing)
- [ ] Updated relevant documentation (`docs/`) or specification (`x/<module>/spec/`)
- [ ] Added relevant `godoc` [comments](https://blog.golang.org/godoc-documenting-go-code).
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Re-reviewed `Files changed` in the Github PR explorer
- [ ] Review `Codecov Report` in the comment section below once CI passes

### CODE REVIEW

**Review by colin-axner — APPROVED** [review]
Looks great to me! Thanks for handling this issue and splitting up the work into several prs (I really appreciate that)

Could you add a changelog entry under "API Breaking" stating that the channel keeper now expects a keeper which fulfills the expected keeper interface for the capability keeper

**Review by crodriguezvega — APPROVED** [review]
Thanks very much, @Anmol1696!

File: modules/core/04-channel/types/expected_keepers.go [hunk 1]
```diff
@@ -76,3 +76,9 @@ type ConnectionKeeper interface {
 type PortKeeper interface {
 	Authenticate(ctx sdk.Context, key *capabilitytypes.Capability, portID string) bool
 }
+
+type ScopedKeeper interface {
```
Comment (damiannolan) [inline modules/core/04-channel/types/expected_keepers.go:80]:
```suggestion

// ScopedKeeper defines the expected x/capability scoped keeper interface
type ScopedKeeper interface {
```

**Review by damiannolan — APPROVED** [review]
Thank you, once again! Appreciate the concise PRs

**Pull request merged**